src/ggrc/models/revision.py

```diff
@@ -318,13 +332,28 @@ def populate_acl(self):
 
     acl_with_people = self._populate_acl_with_people(access_control_list)
     filtered_acl = self._filter_internal_acls(acl_with_people)
+    acl_with_acr = self._populate_acl_with_acr(filtered_acl,
+                                               roles_dict_objects)
+
     result_acl = [
-        acl for acl in filtered_acl if acl["ac_role_id"] in roles_dict
+        acl for acl in acl_with_acr if acl["ac_role_id"] in roles_dict_objects
     ]
     return {
         "access_control_list": result_acl,
     }
 
+  def populate_acr(self):
+    """Add acr model for older revisions of AccessControlList."""
+    if self.resource_type == "AccessControlList":
+      if "ac_role" in self._content:
+        return {}
+      roles_dict_objects = role.get_roles_objects_for(self.resource_type)
```

Comment:
It seems that there should rather be
```suggestion
      roles_dict_objects = role.get_roles_objects_for(self._content["object_type"])
```
since in the ACL's Revision should be added not the ACR of the ACL (which is how it is working now), but the ACR of the ACL's `object_type`. Please fix it.

Reply:
Thank you for the comments, I had run the unit tests and changed as was suggested. 
Could you please re-review?